### ISSUE TITLE
Is the default generated app meant to pass the 'nsp check' command?

### ISSUE THREAD

**Issue by ShiraazMoollatjie** [expressjs/generator#127]
I ran the npm generator and I noticed that if I followed the **Ensure your dependencies are secure** section of the _Security Best Practices_ page, I get 3 vulnerabilities. 

I can understand that this is meant to generate a sample app which will probably change over time, but I am not sure whether the intent is to have the sample app pass the nsp checks as well?

Steps to reproduce:

* run `express my-app`
* then run `nsp check`

versions:

```
express --version && nsp --version
4.13.4
2.6.1
``` 

Output
```
(+) 3 vulnerabilities found
┌───────────────┬───────────────────────────────────────────────────────┐
│               │ Regular Expression Denial of Service                  │
├───────────────┼───────────────────────────────────────────────────────┤
│ Name          │ negotiator                                            │
├───────────────┼───────────────────────────────────────────────────────┤
│ Installed     │ 0.5.3                                                 │
├───────────────┼───────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 0.6.0                                              │
├───────────────┼───────────────────────────────────────────────────────┤
│ Patched       │ >= 0.6.1                                              │
├───────────────┼───────────────────────────────────────────────────────┤
│ Path          │ generated-express-app@0.0.0 > express@4.13.4 > accep… │
├───────────────┼───────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/106                │
└───────────────┴───────────────────────────────────────────────────────┘
┌───────────────┬───────────────────────────────────────────────────────┐
│               │ Regular Expression Denial of Service                  │
├───────────────┼───────────────────────────────────────────────────────┤
│ Name          │ uglify-js                                             │
├───────────────┼───────────────────────────────────────────────────────┤
│ Installed     │ 2.2.5                                                 │
├───────────────┼───────────────────────────────────────────────────────┤
│ Vulnerable    │ <2.6.0                                                │
├───────────────┼───────────────────────────────────────────────────────┤
│ Patched       │ >=2.6.0                                               │
├───────────────┼───────────────────────────────────────────────────────┤
│ Path          │ generated-express-app@0.0.0 > jade@1.11.0 > transfor… │
├───────────────┼───────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/48                 │
└───────────────┴───────────────────────────────────────────────────────┘
┌───────────────┬───────────────────────────────────────────────────────┐
│               │ Incorrect Handling of Non-Boolean Comparisons During… │
├───────────────┼───────────────────────────────────────────────────────┤
│ Name          │ uglify-js                                             │
├───────────────┼───────────────────────────────────────────────────────┤
│ Installed     │ 2.2.5                                                 │
├───────────────┼───────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 2.4.23                                             │
├───────────────┼───────────────────────────────────────────────────────┤
│ Patched       │ >= 2.4.24                                             │
├───────────────┼───────────────────────────────────────────────────────┤
│ Path          │ generated-express-app@0.0.0 > jade@1.11.0 > transfor… │
├───────────────┼───────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/39                 │
└───────────────┴───────────────────────────────────────────────────────┘
```

**Comment by dougwilson** [comment]
Ideally, no. Sadly because I accidentally released some minor version bumps t some Express.js dependencies without also scheduling a new minor of Express.js, we had to make a new emergency minor version bump of Express.js to address a security fix in said dependencies, and we haven't actually gotten around to finishing updating some other things, like this module. I'm working on trying to get the new version of this module finished up and then we'll start generating an app with Express.js 4.14.0 by default, which does not have listed vulnerabilities.

Of the three you have listed:

1. The first will be fixed as soon as we generate with Express 4.14.0 by default.
2. The second two look like they are in the `jade` template module. You can of course always use a different template engine instead of `jade`, but `npm info jade version` shows that 1.11.0 is the most recent version, so I'm not sure if we can do anything there without that module releasing an updated version first.

**Comment by dougwilson** [comment]
The `master` branch has been updated with the fix.
